contracts/TokenVesting.sol

```diff
@@ -2,66 +2,93 @@
 // SPDX-License-Identifier: Apache-2.0
 pragma solidity ^0.8.0;
 
-import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
-import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
-import "@openzeppelin/contracts/utils/math/Math.sol";
-import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "solmate/src/utils/SafeTransferLib.sol";
 
 /**
  * @title TokenVesting
+ * @author Aperture Finance
  */
-contract TokenVesting is Ownable, ReentrancyGuard{
-    using SafeMath for uint256;
-    using SafeERC20 for IERC20;
-    struct VestingSchedule{
+contract TokenVesting is Ownable {
+    using SafeTransferLib for ERC20;
+
+    struct VestingSchedule {
+        // first slot
+        // whether or not the vesting schedule has been initialized
         bool initialized;
-        // beneficiary of tokens after they are released
-        address  beneficiary;
-        // cliff period in seconds
-        uint256  cliff;
-        // start time of the vesting period
-        uint256  start;
-        // duration of the vesting period in seconds
-        uint256  duration;
-        // duration of a slice period for the vesting in seconds
-        uint256 slicePeriodSeconds;
         // whether or not the vesting is revocable
-        bool  revocable;
-        // total amount of tokens to be released at the end of the vesting
-        uint256 amountTotal;
-        // amount of tokens released
-        uint256  released;
+        bool revocable;
         // whether or not the vesting has been revoked
         bool revoked;
+        // beneficiary of tokens after they are released
+        address beneficiary;
+        // cliff in seconds, latest year 2106
+        uint32 cliff;
+        // start time of the vesting period, latest year 2106
+        uint32 start;
+        // second slot
+        // duration of the vesting period in seconds, max 136 years
+        uint32 duration;
+        // total amount of tokens to be released at the end of the vesting
+        uint112 amountTotal;
+        // amount of tokens released
+        uint112 released;
     }
 
     // address of the ERC20 token
-    IERC20 immutable private _token;
+    ERC20 private immutable _token;
```

Comment:
Consider consistent naming for `_token`. What is special about it that we have to add an underscore before it?

Reply:
For privacy.

Reply:
Other private variables don't have `_` prefix?

Reply:
The original author was not consistent.

Reply:
You can.